Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -311,6 +313,20 @@ private func createResolvedPackages(
             }
         }
     }
+    
+    // If a target with similar name was encountered before, we emit a diagnostic.
+    if foundDuplicateTarget {
+        for targetName in allTargetNames.sorted() {
```

Comment:
Instead of using the `allTargetNames` variable here, we can work with `packageBuilders` to find the duplicates targets and their package names. This will give us a cleaner approach where we don't need to rely on an already populated set of target names.

Reply:
@aciidb0mb3r are you thinking we replace the `allTargetNames` with something like `packageBuilders.flatMap({ $0.package.targets.map({ $0.name }) }).sorted()` ?

Reply:
The problem here is that we have to find duplicate items in an array. This is a very generic problem that we can solve using some algorithm. If you look at `CollectionAlgorithm.swift` file, there is a `findDuplicates()` method. We should be able to use that by combining the target list with `KeyedPair` struct. However, it looks like the current implementation of `findDuplicates()` is slightly wrong as it doesn't preserve the first duplicate match.

I think your current solution is good and will work perfectly fine! Lets land that and I'll make a note about switching to a more generalized solution.